src/pal/src/include/pal/file.hpp

```diff
@@ -44,7 +44,8 @@ namespace CorUnix
                                   In Windows we can open a file for writing only */
         int  open_flags;       /* stores Unix file creation flags */
         BOOL open_flags_deviceaccessonly;
-        char unix_filename[MAXPATHLEN];
+        CHAR *unix_filename;
+        DWORD nameSize;
```

Comment:
I would prefer not adding the nameSize here and in the CFileMappingImmutableData. While it gives us a little advantage of not having to measure the string length, it makes the code that copies the name more complex. We can just use strdup to create a copy of the filename that we want to store instead of the malloc + strcpy_s and have only one error path instead of two (we now have one for failed malloc and one for failed strcpy_s).